### PR TITLE
implementation/59434 visualize reminders in notification center

### PR DESCRIPTION
**Pull request by akabiru**
# Ticket

https://community.openproject.org/projects/communicator-stream/work_packages/59434

Follows #17341, #17229

- [x] Expose reminder note to notifications API
  - [ ] Fix notifications eager loading wrapper _(Separate WP https://community.openproject.org/wp/60018)_
- [x] Visualize reminders in notification center. Notification arising from reminders have a note if present
  - [x] Cover notification reason aggregation https://github.com/opf/openproject/pull/17426
- [x] Add Reminders tab in notifications menu

## Screenshots

Screenshot 2024-12-10 at 9 37 31 PM

# What approach did you choose and why?

- Extract "relative time" into a standalone component- reused in actors-inline and reminders components.
Reminders are given first priority in the notification entry footer- so that notes can be displayed.

- Extract reminder notifications from aggregation groups so that they are rendered as standalone and can also be managed as standalone.

# Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation in Lookbook (patterns, previews, etc)
- [ ] Tested major browsers (Chrome, Firefox, Edge, ...)

### CODE REVIEW

**Review by oliverguenther — APPROVED** [review]
LGTM code-wise, I have not checked or tested the AC

File: frontend/src/app/features/in-app-notifications/entry/reminder-alert/in-app-notification-reminder-alert.component.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<op-in-app-notification-relative-time
+  [notification]="reminderAlert"
+  [hasActorByLine]="false"
+/>
+<span
```
Comment (jjabari-op) [inline frontend/src/app/features/in-app-notifications/entry/reminder-alert/in-app-notification-reminder-alert.component.html:5]: Bildschirmfoto 2024-12-11 um 13 25 51

I still see the issue raised by wieland in our last meeting as well - a longer note text gets hidden on mobile

**Review by jjabari-op — APPROVED** [review]
Great work @akabiru 🎉 

I didn't find any code related issues. Only UI wise I saw the mobile issue already described in our last meeting. Still approving it!

File: frontend/src/app/features/in-app-notifications/entry/reminder-alert/in-app-notification-reminder-alert.component.html (hunk 1 above)
Comment (akabiru) [reply frontend/src/app/features/in-app-notifications/entry/reminder-alert/in-app-notification-reminder-alert.component.html:5]: Thanks! I had planned on resolving separately in https://community.openproject.org/wp/59989 but it should also be small enough to fit here 👍🏾

Comment (akabiru) [reply frontend/src/app/features/in-app-notifications/entry/reminder-alert/in-app-notification-reminder-alert.component.html:5]: https://github.com/opf/openproject/pull/17323/commits/0ce5623e7a7ab4e63b62d0abbf9ad86fd65264d2

Screenshot 2024-12-12 at 11 18 37 AM

**Pull request merged**